atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunner.java

```diff
@@ -15,19 +15,361 @@
+    private static final Logger log = LoggerFactory.getLogger(SweepTaskRunner.class);
+
+    private final KeyValueService keyValueService;
+    private final LongSupplier unreadableTimestampSupplier;
+    private final LongSupplier immutableTimestampSupplier;
+    private final TransactionService transactionService;
+    private final SweepStrategyManager sweepStrategyManager;
+    private final CellsSweeper cellsSweeper;
+
+    public SweepTaskRunner(
+            KeyValueService keyValueService,
+            LongSupplier unreadableTimestampSupplier,
+            LongSupplier immutableTimestampSupplier,
+            TransactionService transactionService,
+            SweepStrategyManager sweepStrategyManager,
+            CellsSweeper cellsSweeper) {
+        this.keyValueService = keyValueService;
+        this.unreadableTimestampSupplier = unreadableTimestampSupplier;
+        this.immutableTimestampSupplier = immutableTimestampSupplier;
+        this.transactionService = transactionService;
+        this.sweepStrategyManager = sweepStrategyManager;
+        this.cellsSweeper = cellsSweeper;
+    }
 
-public interface SweepTaskRunner {
     /**
-     * Performs a dry run of sweep, doing all of the reading, but not deleting data.
+     * Represents the type of run to be conducted by the sweep runner.
      */
-    SweepResults dryRun(TableReference tableRef, int rowBatchSize, int cellBatchSize, @Nullable byte[] startRow);
+    private enum RunType {
+        /**
+         * A DRY run is expect to not mutate any tables (with the exception of the sweep.progress table)
+         * but will determine and report back all the values that *would* have been swept.
+         */
+        DRY,
+
+        /**
+         * A FULL run will execute all followers / sentinel additions / and deletions on the
+         * cells that qualify for sweeping.
+         */
+        FULL
+    }
+
+    public SweepResults dryRun(TableReference tableRef,
+            int rowBatchSize,
+            int cellBatchSize,
+            @Nullable byte[] startRow) {
+        return runInternal(tableRef, rowBatchSize, cellBatchSize, startRow, RunType.DRY);
+    }
+
+    public SweepResults run(TableReference tableRef, int rowBatchSize, int cellBatchSize, @Nullable byte[] startRow) {
+        return runInternal(tableRef, rowBatchSize, cellBatchSize, startRow, RunType.FULL);
+    }
+
+    public long getConservativeSweepTimestamp() {
+        return Sweeper.CONSERVATIVE.getSweepTimestampSupplier().getSweepTimestamp(
+                unreadableTimestampSupplier, immutableTimestampSupplier);
+    }
+
+    private SweepResults runInternal(
+            TableReference tableRef,
+            int rowBatchSize,
+            int cellBatchSize,
+            @Nullable byte[] nullableStartRow,
+            RunType runType) {
+        Preconditions.checkNotNull(tableRef, "tableRef cannot be null");
+        Preconditions.checkState(!AtlasDbConstants.hiddenTables.contains(tableRef));
```

Comment:
The diff is not being displayed correctly. This class is in fact a rename of SweepTaskRunnerImpl and didn't touch this.